### ISSUE TITLE
Training ProstT5 MLM to predict full sequence or labels

### ISSUE THREAD

**Issue by BSharmi** [mheinzinger/ProstT5#32]
Hi there!

I have been using ProtT5 (your other piece of amazing work!) for a while now and remember that ProtT5 was trained BERT style to predict full sequences instead of predicting labels only. For ProstT5 I was going through the pre-training LLM code (https://github.com/mheinzinger/ProstT5/blob/main/scripts/pretraining_scripts/pretraining_stage1_MLM.py#L338C33-L338C47) 
I noticed the data processing is same as the original Flan T5. If that is right why would we expect for the generated sequence to be the same as ground truth (https://github.com/mheinzinger/ProstT5/blob/main/scripts/translate.py#L221)? Or did I miss something here in the generation?

Would really appreciate any pointers!

Thank you!

**Comment by mheinzinger** [comment]
Hi, thanks for your kind words :)
Indeed, for ProstT5, I wanted to assess whether we can also pre-train using T5's original span-denoising (we failed to make it work for original ProtT5 which is why we went for BERT-style masking..).
I noticed that ProstT5 trained fine with span-denoising so I sticked to it because I consider it the more complicated task as the model also has to learn how many tokens to predict.
However, the pre-training consisted of two steps: first step was the span-denoising for teaching the model to "read" the new 3Di tokens. Second step, was then the actual translation between sequence and structure.
The evaluation script you are referring to, evaluates only the checkpoint after the second step pre-training (after training to translate between sequence and structure).
Hope this helps; let me know if sth remains unclear!
